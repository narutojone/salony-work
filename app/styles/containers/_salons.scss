.image {
    width: 50px;
    height: 50px;
}

.margintop {
    margin-top: 50px;
}

.secondtable th {
    display: none;
}

.lazyother {
    margin-top: -21px;
}

.lazysecond {
    margin-top: -30px;
}