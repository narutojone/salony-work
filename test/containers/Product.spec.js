
import React from 'react';
import { shallow } from 'enzyme';
import { ProductContainer } from 'containers/Product';

const mockDispatch = jest.fn();

const props = {
  app: {
    alerts: [],
    app: {
      data: {
        product: {
          images: [],
        },
      },
    },
  },
  dispatch: mockDispatch,
  user: {
    isAuthenticated: false,
  },
};

function setup(ownProps = props) {
  return shallow(
    <ProductContainer {...ownProps} />,
    { attachTo: document.getElementById('react') }
  );
}

describe('ProductContainer', () => {
  const wrapper = setup();

  it('should be a Component', () => {
    expect(wrapper.instance() instanceof React.Component).toBe(true);
  });

  it('should render properly for anonymous users', () => {
    expect(wrapper.find('ImageGallery')).toBePresent();
    expect(wrapper.find('HourTable')).toBePresent();
    expect(wrapper.find('BootstrapTable')).toBePresent();
    expect(wrapper.find('TableHeaderColumn')).toBePresent();
  });

  it('should render properly', () => {
    expect(wrapper.find('.container')).toBePresent();
    expect(wrapper.find('.slider')).toBePresent();
    expect(wrapper.find('.margintop')).toBePresent();
    expect(wrapper.find('.table')).toBePresent();
    expect(wrapper.find('.description')).toBePresent();
  });

  it('should have dispatched an action on mount', () => {
    expect(mockDispatch.mock.calls[0][0]).toEqual({
      payload: { query: undefined },
      type: 'GET_PRODUCTBYID',
    });
  });

  it('should render some items when data arrives', () => {
    wrapper.setProps({
      app: {
        app: {
          data: {
            product: {
              images: [],
            },
          },
        },
      },
    });
    expect(wrapper.find('.container')).toMatchSnapshot();
  });

  it('should have dispatched an action on mount', () => {
    expect(mockDispatch.mock.calls[1][0]).toEqual({
      payload: { query: undefined },
      type: 'GET_SERVICESBYID',
    });
  });
});
