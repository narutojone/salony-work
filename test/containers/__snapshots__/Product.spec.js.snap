// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductContainer should render some items when data arrives 1`] = `
<div
  className="container"
>
  <div>
    <h2 />
  </div>
  <div
    className="slider margintop"
  >
    <ImageGallery
      autoPlay={false}
      disableArrowKeys={false}
      disableSwipe={false}
      disableThumbnailScroll={false}
      flickThreshold={0.4}
      indexSeparator=" / "
      infinite={true}
      items={
        Array [
          Object {
            "original": "http://www.cmp-cyprus.org/sites/all/modules/media_gallery/images/empty_gallery.png",
          },
        ]
      }
      lazyLoad={false}
      preventDefaultTouchmoveEvent={false}
      renderFullscreenButton={[Function]}
      renderLeftNav={[Function]}
      renderPlayPauseButton={[Function]}
      renderRightNav={[Function]}
      showBullets={false}
      showFullscreenButton={false}
      showIndex={false}
      showNav={true}
      showPlayButton={false}
      showThumbnails={false}
      sizes="300"
      slideDuration={450}
      slideInterval={3000}
      slideOnThumbnailHover={false}
      startIndex={0}
      stopPropagation={false}
      swipeThreshold={30}
      swipingTransitionDuration={0}
      thumbnailPosition="bottom"
      useBrowserFullscreen={true}
      useTranslate3D={true}
    />
  </div>
  <div
    className="margintop"
  >
    <h5>
      Working hours
    </h5>
    <HourTable />
  </div>
  <div
    className="margintop"
  >
    <h5>
      Description
    </h5>
    <h5
      className="description"
    />
    <h5
      className="margintop"
    >
      Service
    </h5>
    <BootstrapTable
      autoCollapse={
        Object {
          "filter": false,
          "search": false,
          "sort": false,
        }
      }
      bodyContainerClass={null}
      bordered={true}
      cellEdit={
        Object {
          "afterSaveCell": undefined,
          "beforeSaveCell": undefined,
          "blurToEscape": false,
          "blurToSave": false,
          "mode": "none",
          "nonEditableRows": undefined,
        }
      }
      className="table"
      columnFilter={false}
      condensed={false}
      containerClass={null}
      csvFileName="spreadsheet.csv"
      deleteRow={false}
      expandColumnOptions={
        Object {
          "expandColumnBeforeSelectColumn": true,
          "expandColumnComponent": undefined,
          "expandColumnVisible": false,
          "expandedColumnHeaderComponent": undefined,
        }
      }
      exportCSV={false}
      fetchInfo={
        Object {
          "dataTotalSize": 0,
        }
      }
      footer={false}
      headerContainerClass={null}
      height="100%"
      hover={true}
      ignoreSinglePage={false}
      insertRow={false}
      keyBoardNav={false}
      multiColumnSearch={false}
      multiColumnSort={1}
      options={
        Object {
          "afterColumnFilter": undefined,
          "afterDeleteRow": undefined,
          "afterInsertRow": undefined,
          "afterSearch": undefined,
          "afterTableComplete": undefined,
          "alwaysShowAllBtns": false,
          "beforeShowError": undefined,
          "btnGroup": undefined,
          "clearSearch": false,
          "clearSearchBtn": undefined,
          "closeText": "Close",
          "defaultSearch": "",
          "defaultSortName": undefined,
          "defaultSortOrder": undefined,
          "deleteBtn": undefined,
          "deleteText": "Delete",
          "excludeCSVHeader": false,
          "expandAll": false,
          "expandBodyClass": null,
          "expandBy": "row",
          "expandParentClass": null,
          "expandRowBgColor": undefined,
          "expanding": Array [],
          "exportCSVBtn": undefined,
          "exportCSVSeparator": ",",
          "exportCSVText": "Export to CSV",
          "firstPage": "<<",
          "firstPageTitle": "first page",
          "handleConfirmDeleteRow": undefined,
          "hidePageListOnlyOnePage": false,
          "hideRowOnExpand": false,
          "hideSizePerPage": false,
          "ignoreEditable": false,
          "insertBtn": undefined,
          "insertFailIndicator": "Validation errors, please check!",
          "insertModal": undefined,
          "insertModalBody": undefined,
          "insertModalFooter": undefined,
          "insertModalHeader": undefined,
          "insertText": "New",
          "keepSizePerPageState": false,
          "lastPage": ">>",
          "lastPageTitle": "last page",
          "nextPage": ">",
          "nextPageTitle": "next page",
          "noAutoBOM": true,
          "noDataText": undefined,
          "onExpand": undefined,
          "onMouseEnter": undefined,
          "onMouseLeave": undefined,
          "onRowClick": undefined,
          "onRowDoubleClick": undefined,
          "onRowMouseOut": undefined,
          "onRowMouseOver": undefined,
          "onSizePerPageList": undefined,
          "onlyOneExpanding": false,
          "page": undefined,
          "pageStartIndex": 1,
          "paginationPanel": undefined,
          "paginationPosition": "bottom",
          "paginationShowsTotal": false,
          "paginationSize": 5,
          "prePage": "<",
          "prePageTitle": "previous page",
          "printToolBar": true,
          "saveText": "Save",
          "searchDelayTime": undefined,
          "searchField": undefined,
          "searchPanel": undefined,
          "searchPosition": "right",
          "showSelectedOnlyBtn": undefined,
          "sizePerPage": undefined,
          "sizePerPageDropDown": undefined,
          "sizePerPageList": Array [
            10,
            25,
            30,
            50,
          ],
          "sortIndicator": true,
          "sortName": undefined,
          "sortOrder": undefined,
          "toolBar": undefined,
          "toolbarPosition": "top",
          "withFirstAndLast": true,
          "withoutNoDataText": false,
        }
      }
      pagination={false}
      printable={true}
      renderAlert={true}
      replace={false}
      search={false}
      selectRow={
        Object {
          "bgColor": "",
          "clickToExpand": false,
          "clickToSelect": false,
          "clickToSelectAndEditCell": false,
          "customComponent": undefined,
          "hideSelectColumn": false,
          "mode": "none",
          "onSelect": undefined,
          "onSelectAll": undefined,
          "onlyUnselectVisible": false,
          "selected": Array [],
          "showOnlySelected": false,
          "unselectable": Array [],
        }
      }
      striped={true}
      tableBodyClass={null}
      tableContainerClass={null}
      tableFooterClass={null}
      tableHeaderClass={null}
      trClassName=""
      version="3"
      withoutTabIndex={false}
    >
      <TableHeaderColumn
        className=""
        columnClassName=""
        columnTitle={false}
        csvFieldType="string"
        dataAlign="left"
        dataField="name"
        dataSort={true}
        defaultASC={false}
        editColumnClassName=""
        editable={true}
        expandable={true}
        filterFormatted={false}
        headerTitle={true}
        hidden={false}
        hiddenOnInsert={false}
        invalidEditColumnClassName=""
        isKey={true}
        keyValidator={false}
        searchable={true}
        sortIndicator={true}
        width="400"
      >
        Name
      </TableHeaderColumn>
      <TableHeaderColumn
        className=""
        columnClassName=""
        columnTitle={false}
        csvFieldType="string"
        dataAlign="left"
        dataField="duration"
        dataSort={true}
        defaultASC={false}
        editColumnClassName=""
        editable={true}
        expandable={true}
        filterFormatted={false}
        headerTitle={true}
        hidden={false}
        hiddenOnInsert={false}
        invalidEditColumnClassName=""
        isKey={false}
        keyValidator={false}
        searchable={true}
        sortIndicator={true}
        width="400"
      >
        Duration (Min)
      </TableHeaderColumn>
      <TableHeaderColumn
        className=""
        columnClassName=""
        columnTitle={false}
        csvFieldType="string"
        dataAlign="left"
        dataField="price"
        dataSort={true}
        defaultASC={false}
        editColumnClassName=""
        editable={true}
        expandable={true}
        filterFormatted={false}
        headerTitle={true}
        hidden={false}
        hiddenOnInsert={false}
        invalidEditColumnClassName=""
        isKey={false}
        keyValidator={false}
        searchable={true}
        sortIndicator={true}
        width="400"
      >
        Price (KD)
      </TableHeaderColumn>
    </BootstrapTable>
  </div>
</div>
`;
